src/prisma-service.ts

```diff
@@ -0,0 +1,956 @@
+    if (!metagraph_id || !address)
+      return missingParameterResponse('identifier or address');
+
+    const where = { where: { destination_addr: address } };
+
+    return metagraphTransactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencyBalanceByAddress = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id, address } = event.pathParameters || {};
+    if (!metagraph_id || !address)
+      return missingParameterResponse('identifier or address');
+
+    const balance = await prisma.metagraph_balance_changes.findFirst({
+      where: {
+        metagraph_id,
+        address
+      },
+      orderBy: {
+        metagraph_snapshot_ordinal: 'desc'
+      }
+    });
+
+    return respond(balance, balanceResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencyFeeTransaction = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id, hash } = event.pathParameters || {};
+    if (!metagraph_id || !hash)
+      return missingParameterResponse('identifier or hash');
+
+    const transaction = await prisma.metagraph_fee_transactions.findUnique({
+      where: {
+        metagraph_id_hash: {
+          metagraph_id,
+          hash
+        }
+      },
+      include: {
+        metagraph_snapshots: { select: { hash: true, ordinal: true } }
+      }
+    });
+
+    return respond(transaction, metagraphFeeTransactionResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+const metagraphFeeTransactionsQuery = async (
+  baseQuery,
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const query = {
+      ...baseQuery,
+      include: {
+        metagraph_snapshots: { select: { hash: true, ordinal: true } }
+      },
+      orderBy: [
+        { created_at: Prisma.SortOrder.desc },
+        { ordinal: Prisma.SortOrder.desc }
+      ]
+    };
+
+    const toCursor = (row) => ({
+      metagraph_id: row.metagraph_id,
+      hash: row.hash
+    });
+    const fromCursor = (row) => ({
+      metagraph_id: row.metagraph_id,
+      hash: row.hash
+    });
+
+    return await paginatedResults(
+      event,
+      toCursor,
+      fromCursor,
+      query,
+      prisma.metagraph_fee_transactions.findMany,
+      metagraphFeeTransactionsResponse
+    );
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencySnapshotFeeTransactions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id, term } = event.pathParameters || {};
+    if (!metagraph_id || !term)
+      return missingParameterResponse('identifier or term');
+
+    const where = {
+      metagraph_id: metagraph_id,
+      ordinal: BigInt(term),
+      fee: { gt: 0 }
```

Comment:
FeeTransactions don't have a `fee` field so this probably shouldn't be here. We don't want to filter only snapshots that have a fee, although that would cover all snapshots.